src/mono/mono/mini/simd-intrinsics.c

```diff
@@ -306,6 +306,19 @@ emit_simd_ins_for_binary_op (MonoCompile *cfg, MonoClass *klass, MonoMethodSigna
 				break;
 			case SN_Multiply:
 			case SN_op_Multiply:
+				if (fsig->params [1]->data.array == NULL) {
```

Comment:
This isn't a good way to check if something is not an array.

```c
static gboolean 
type_is_array (MonoType *type)
{
	return  type->type == MONO_TYPE_ARRAY  || type->type == MONO_TYPE_SZARRAY;
}
```

and then `!type_is_array (fsig->params [1])`

Also in a few other places.

Reply:
Fixed.